src/util/saveArticles.ts

```diff
@@ -247,7 +247,7 @@ export function getArticleUrl(downloader: Downloader, dump: Dump, articleId: str
 /*
  * Fetch Articles
  */
-export async function saveArticles(zimCreator: ZimCreator, downloader: Downloader, redisStore: RS, dump: Dump) {
+export async function saveArticles(zimCreator: ZimCreator, downloader: Downloader, redisStore: RS, dump: Dump, forceRender = null) {
```

Comment:
This looks wrong to me, we should transmit the renders objects directly

Reply:
We need to keep `forceRender` argument in saveArticles. When run the `execute()` function, we grab `forceRender` from params and pass it to `saveArticles()` because we instantiate the renderer builder with `auto` or `specific` render type option there. Or you suggest to instantiate it in `src/mwoffliner.lib.ts` directly and pass entire `mainPageRenderer` and `articlesRenderer` into saveArticles()?

Reply:
IMO we face here again the problematic around bad architecture of `saveArticles()`. Skipping for the moment.